From 4a3f2fa9ed7260ed0d8fca1aa6130521d8cedb49 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Fri, 23 Jan 2015 15:34:14 +0100
Subject: [PATCH 3/7] Adding a CRLF-file with core.autocrlf=input and
 core.safecrlf=true does not fail as with vanilla git

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/crlf.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/crlf.c b/src/crlf.c
index c1f87485e..d88ad00f3 100644
--- a/src/crlf.c
+++ b/src/crlf.c
@@ -160,6 +160,19 @@ static int crlf_apply_to_odb(
 			default:
 				break;
 			}
+		} else if (stats.crlf && ca->auto_crlf == GIT_AUTO_CRLF_INPUT) {
+			switch (ca->safe_crlf) {
+			case GIT_SAFE_CRLF_FAIL:
+				giterr_set(
+					GITERR_FILTER, "CRLF would be replaced by LF in '%s'",
+					git_filter_source_path(src));
+				return -1;
+			case GIT_SAFE_CRLF_WARN:
+				/* TODO: issue warning when warning API is available */;
+				break;
+			default:
+				break;
+			}
 		}
 
 		/*
-- 
2.16.1.windows.4

